qiskit_machine_learning/kernels/quantum_kernel.py

```diff
@@ -223,106 +238,108 @@ def evaluate(self, x_vec: np.ndarray, y_vec: np.ndarray = None) -> np.ndarray:
                     "x_vec and class feature map have incompatible dimensions.\n" +
                     "x_vec has %s dimensions, but feature map has %s." %
                     (x_vec.shape[1], self._feature_map.num_parameters)) from AttributeError
-
         if y_vec is not None and y_vec.shape[1] != self._feature_map.num_parameters:
             raise ValueError("y_vec and class feature map have incompatible dimensions.\n" +
                              "y_vec has %s dimensions, but feature map has %s." %
                              (y_vec.shape[1], self._feature_map.num_parameters))
```

Comment:
Is there a reason why the blocks for `x_vec` and `y_vec` are slightly different?

Reply:
@dlasecki I don't really get the question. 
If you mean this block: https://github.com/Qiskit/qiskit-machine-learning/blob/master/qiskit_machine_learning/kernels/quantum_kernel.py#L218:L230 then it looks good to me.

Reply:
Sorry, my line selection did not render well. I was hoping to remove some code duplication there and have been wondering about the missing `try` for `y_vec`. But now I noticed the previous `if  `statement that explains it.